test/e2e/framework/kubernetes/get-pod-ip.go

```diff
@@ -0,0 +1,28 @@
+package kubernetes
+
+import (
+	"context"
+
+	"github.com/pkg/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/tools/clientcmd"
+)
+
+func GetPodIP(kubeConfigFilePath, namespace, podName string) (string, error) {
+	config, err := clientcmd.BuildConfigFromFlags("", kubeConfigFilePath)
+	if err != nil {
+		return "", errors.Wrapf(err, "error building kubeconfig")
+	}
+
+	clientset, err := kubernetes.NewForConfig(config)
+	if err != nil {
+		return "", errors.Wrapf(err, "error creating Kubernetes clientset")
+	}
+
+	pod, err := clientset.CoreV1().Pods(namespace).Get(context.Background(), podName, metav1.GetOptions{})
+	if err != nil {
+		return "", errors.Wrapf(err, "error getting pod %s in namespace %s", podName, namespace)
+	}
+	return pod.Status.PodIP, nil
```

Comment:
Depending on what point this method is called, PodIP might be nill as the IP has not been allocated. Most likely not an issue.

Reply:
right now, im only using it in steps after we check that agnhost pod is up and running, but i can add an assert not empty check to make sure